build_tools/github_actions/tests/configure_ci_test.py

```diff
@@ -106,17 +102,14 @@ def test_valid_linux_pull_request_matrix_generator(self):
             families={},
             platform="linux",
         )
-        linux_target_output.sort(key=lambda item: item["family"])
-        linux_target_to_compare = [
-            {"test-runs-on": "", "family": "gfx110X-dgpu", "pytorch-target": "gfx1100"},
-            {
-                "test-runs-on": "linux-mi300-1gpu-ossci-rocm",
-                "family": "gfx94X-dcgpu",
-                "pytorch-target": "gfx942",
-            },
-        ]
-        linux_target_to_compare.sort(key=lambda item: item["family"])
-        self.assertEqual(linux_target_output, linux_target_to_compare)
+        self.assertTrue(
+            any("gfx94X-dcgpu" == entry["family"] for entry in linux_target_output)
+        )
+        self.assertTrue(
+            any("gfx110X-dgpu" == entry["family"] for entry in linux_target_output)
+        )
+        self.assertEqual(len(linux_target_output), 2)
+        self.assertTrue(all("family" in entry for entry in linux_target_output))
```

Comment:
Another option would be to add a test helper like `assert_target_output_is_valid` that runs through a standard set of checks, then just call that in each test case. You could then add additional checks _on top of that_ where it's worth calling out the unique behavior of each configuration.